rest-server/engines/src/main/scala/com/actionml/engines/ur/URAlgorithm.scala

```diff
@@ -558,11 +558,23 @@ class URAlgorithm private (
     }
   }
 
-  private def getItemSetMatchers(query: URQuery): Seq[Matcher] = {
-    query.itemSet.getOrElse(Seq.empty).map(item => Matcher(
-      modelEventNames.head, // only look for items that have been converted on
-      query.itemSet.getOrElse(Seq.empty),
-      params.itemSetBias))
+
+  private def getItemSetMatchers(query: URQuery)(implicit ec: ExecutionContext): Future[Seq[Matcher]] = {
+    if(query.itemSet.getOrElse(Seq.empty).size != 1){
+      Future(query.itemSet.getOrElse(Seq.empty).map(item => Matcher(
+        modelEventNames.head, // only look for items that have been converted on
+        query.itemSet.getOrElse(Seq.empty),
+        params.itemSetBias))
+      )
+    } else { // one item in the set so better to use an item query
+      val newBias = if(query.itemSetBias.nonEmpty) query.itemSetBias else params.itemSetBias
+      getSimilarItemsMatchers(
```

Comment:
switching to getSimilarItemsMatchers when only one item in the itemSet. Had to make getItemSetMatchers return Future[Seq[Matcher]]  so that both paths return the same thing, namely a Future. No need for a Future in the "if" part but should not hurt anything either, right?